econml/metalearners.py

```diff
@@ -117,6 +116,13 @@ class SLearner(BaseCateEstimator):
 
     def __init__(self, overall_model):
         self.overall_model = clone(overall_model, safe=False)
+        self._label_encoder = LabelEncoder()
+        self._one_hot_encoder = OneHotEncoder(categories='auto', sparse=False)
+        self.transformer = FunctionTransformer(
+            func=(lambda T:
+                  self._one_hot_encoder.transform(
+                      self._label_encoder.transform(T).reshape(-1, 1))[:, 1:]),
+            validate=False)
         super().__init__()
 
     @BaseCateEstimator._wrap_fit
```

Comment:
Consider making X optional (and treating it as a 0-column array if it is None).

Also, make inference keyword only.